File: app/src/main/java/com/jawnnypoo/openmeh/data/Item.java
package com.jawnnypoo.openmeh.data;

import android.support.annotation.NonNull;

import org.parceler.Parcel;

import java.util.List;

/**
 * An item of the deal
 * Created by John on 4/17/2015.
 */
@Parcel
public class Item implements Comparable<Item>{

    List<Attribute> attributes;
    String condition;
    String id;
    Float price;
    String photo;

    public Item(){}

    public List<Attribute> getAttributes() {
        return attributes;
    }

    public String getCondition() {
        return condition;
    }

    public String getId() {
        return id;
    }

    public Float getPrice() {
        return price;
    }

    public String getPhoto() {
        return photo;
    }

    @Override
    public int compareTo(@NonNull Item another) {
        return Float.compare(this.getPrice(), another.getPrice());
    }
}

File: shared/src/main/java/com/jawnnypoo/openmeh/shared/model/Item.kt
package com.jawnnypoo.openmeh.shared.model

import com.squareup.moshi.Json

/**
 * An item of the deal
 */
data class Item(
        @Json(name = "id") var id: String? = null,
        @Json(name = "price") var price: Float? = null
)
